diff --git a/internal/multicluster/internal/types/exported_services_test.go b/internal/multicluster/internal/types/exported_services_test.go
index 6af8e2c779..77efc276f8 100644
--- a/internal/multicluster/internal/types/exported_services_test.go
+++ b/internal/multicluster/internal/types/exported_services_test.go
@@ -34,7 +34,7 @@ func validExportedServicesWithPartition() *multiclusterv1alpha1.ExportedServices
 	consumers := []*multiclusterv1alpha1.ExportedServicesConsumer{
 		{
 			ConsumerTenancy: &multiclusterv1alpha1.ExportedServicesConsumer_Partition{
-				Partition: "default",
+				Partition: "partition",
 			},
 		},
 	}
@@ -48,7 +48,7 @@ func validExportedServicesWithSamenessGroup() *multiclusterv1alpha1.ExportedServ
 	consumers := []*multiclusterv1alpha1.ExportedServicesConsumer{
 		{
 			ConsumerTenancy: &multiclusterv1alpha1.ExportedServicesConsumer_SamenessGroup{
-				SamenessGroup: "",
+				SamenessGroup: "sameness_group",
 			},
 		},
 	}
diff --git a/internal/multicluster/internal/types/namespace_exported_services_test.go b/internal/multicluster/internal/types/namespace_exported_services_test.go
index d4f2249db6..075e75ff8b 100644
--- a/internal/multicluster/internal/types/namespace_exported_services_test.go
+++ b/internal/multicluster/internal/types/namespace_exported_services_test.go
@@ -27,7 +27,7 @@ func validNamespaceExportedServicesWithPartition() *multiclusterv1alpha1.Namespa
 	consumers := []*multiclusterv1alpha1.ExportedServicesConsumer{
 		{
 			ConsumerTenancy: &multiclusterv1alpha1.ExportedServicesConsumer_Partition{
-				Partition: "default",
+				Partition: "partition",
 			},
 		},
 	}
diff --git a/internal/multicluster/internal/types/partition_exported_services_test.go b/internal/multicluster/internal/types/partition_exported_services_test.go
index 0989ddc83b..0a9732b916 100644
--- a/internal/multicluster/internal/types/partition_exported_services_test.go
+++ b/internal/multicluster/internal/types/partition_exported_services_test.go
@@ -27,7 +27,7 @@ func validPartitionExportedServicesWithPartition() *multiclusterv1alpha1.Partiti
 	consumers := []*multiclusterv1alpha1.ExportedServicesConsumer{
 		{
 			ConsumerTenancy: &multiclusterv1alpha1.ExportedServicesConsumer_Partition{
-				Partition: "default",
+				Partition: "partition",
 			},
 		},
 	}
